src/sentry/integrations/github/client.py

```diff
@@ -224,6 +224,25 @@ def get_commit(self, repo: str, sha: str) -> Any:
         """
         return self.get_cached(f"/repos/{repo}/commits/{sha}")
 
+    def get_merge_commit_sha_from_commit(self, repo: str, sha: str) -> str | None:
+        """
+        Get the merge commit sha from a commit sha.
+        """
+        response = self.get_pullrequest_from_commit(repo, sha)
+        if not isinstance(response, list) or len(response) != 1:
+            # the response should return a single merged PR, return if multiple
+            if len(response) > 1:
+                return None
```

Comment:
why do we need this check if the parent if already does this check?

Reply:
what do you mean by parent? i isolated the logic for getting the github pull request from the commit in this function

Reply:
oh you mean the if above, you're right